src/components/Service/TestService.jsx

```diff
@@ -1,59 +1,164 @@
-    if (testTextAreaRef.current) {
-      testTextAreaRef.current.addEventListener("scroll", updateScroll);
+    if (currentContentIndex < contents.length) {
+      setStartTime(Date.now());
+
+      const minReadTime = calculateMinReadTime(
+        contents[currentContentIndex].content,
+      );
+
+      setCountdown(minReadTime);
+      setCurrentTitle(CONTENTS.title.test[contents[currentContentIndex].type]);
+
+      const timer = setInterval(updateCountdown, 1000);
+
+      return () => {
+        clearInterval(timer);
+      };
+    }
+
+    return () => {};
+  }, [currentContentIndex, contents, updateCountdown]);
+
+  const handleNextContent = () => {
+    const endTime = Date.now();
+    const readingTime = (endTime - startTime) / 1000;
+
+    setReadingTimes([
+      ...readingTimes,
+      { contentId: currentContentIndex, time: readingTime },
+    ]);
+
+    if (currentContentIndex < contents.length - 1) {
+      setCurrentContentIndex(currentContentIndex + 1);
+      setButtonEnabled(false);
+    } else {
+      navigateNextPage("test");
     }
+  };
+
+  const progress = ((currentContentIndex + 1) / contents.length) * 100;
+
+  const renderCodeTags = (str) => {
+    const tags = str.split(/(<code>.*?<\/code>)/);
 
-    return () => {
-      if (testTextAreaRef.current) {
-        testTextAreaRef.current.removeEventListener("scroll", updateScroll);
+    return tags.map((part) => {
+      if (part.startsWith("<code>") && part.endsWith("</code>")) {
+        const codeContent = part.slice(6, -7);
+
+        return (
+          <code key={crypto.randomUUID()} className="code code-inline">
+            {codeContent.replace(/&lt;/g, "<").replace(/&gt;/g, ">")}
+          </code>
+        );
       }
-    };
-  }, []);
+
+      return part.replace(/&lt;/g, "<").replace(/&gt;/g, ">");
+    });
+  };
+
+  const renderContent = (content) => {
+    if (content.type === "code") {
+      const highlightedCode = codeHighlighter({ code: content.content });
+
+      if (highlightedCode === null) {
+        return <p>코드를 표시할 수 없습니다.</p>;
+      }
+
+      return (
+        <pre className="p-4 overflow-x-auto text-sm bg-gray-100 rounded-md">
+          <code className="code-block">{highlightedCode}</code>
+        </pre>
+      );
+    }
+    return <p>{renderCodeTags(content.content)}</p>;
+  };
+
+  const customButtonArea = (
+    <div className="flex items-center justify-between w-4/5 m-auto mt-4 max-mobile:w-full">
+      <p className="m-0 mt-auto text-base text-gray">
+        {countdown > 0
+          ? `다음 글까지 ${countdown}초 남았어요.`
+          : "다음 버튼을 눌러주세요."}
+      </p>
+      <button
+        onClick={handleNextContent}
+        disabled={!buttonEnabled}
+        data-test="test-next-button"
+        className={
+          buttonEnabled
+            ? "px-4 py-2 text-white bg-dark-gray rounded-lg hover:bg-black max-mobile:text-sm"
+            : "px-4 py-2 text-white bg-medium-gray rounded-lg max-mobile:text-sm"
+        }
```

Comment:
```suggestion
        className={`px-4 py-2 text-white rounded-lg max-mobile:text-sm ${buttonEnabled ? "bg-dark-gray hover:bg-black" : "bg-medium-gray"}`}
```
코드의 중복을 줄일 수 있을 것 같습니다.